src/maps/scotland/ferries_connections.ts

```diff
@@ -0,0 +1,69 @@
+import z from "zod";
+import { ClaimAction, ClaimData } from "../../engine/build/claim";
+import { ConnectCitiesAction } from "../../engine/build/connect_cities";
+import { CoordinatesZod } from "../../utils/coordinates";
+import { City } from "../../engine/map/city";
+import { Grid } from "../../engine/map/grid";
+import { Land } from "../../engine/map/location";
+import { TOWN, Track } from "../../engine/map/track";
+import { assert } from "../../utils/validate";
+import { BuildAction, BuildData } from "../../engine/build/build";
+
+export class ScotlandClaimAction extends ClaimAction {
+  validate(data: ClaimData): void {
+    super.validate(data);
+    const land = this.grid().get(data.coordinates) as Land;
+    const track = land.getTrack().find((track) => track.isClaimable())!;
+    assert(canClaim(this.grid(), track), {
+      invalidInput:
+        "Cannot claim track without urbanizing both of connected towns first.",
+    });
+  }
+}
+
+function canClaim(grid: Grid, track: Track) {
+  return track.getExits().every((exit) => {
+    if (exit === TOWN) return false;
+    const neighbor = grid.get(track.coordinates.neighbor(exit));
+    if (neighbor == null) return false;
+    if (neighbor instanceof City) return true;
+  });
+}
+
+export const ConnectCitiesData = z.object({
+  connect: CoordinatesZod.array(),
+});
+
+export type ConnectCitiesData = z.infer<typeof ConnectCitiesData>;
+
+export class ScotlandConnectCitiesAction extends ConnectCitiesAction {
+  validate(data: ConnectCitiesData): void {
```

Comment:
Is there a reason you don't call super.validate? It seems to replicate the existing getMaxBuilds check.

Reply:
In fact, I'm not sure I'm seeing what's different about this action and the regular implementation.